contrib/mod_tls.c

```diff
@@ -10428,38 +10381,20 @@ static int tls_verify_cb(int ok, X509_STORE_CTX *ctx) {
 #endif /* OpenSSL-1.1.x and later */
 
     switch (ctx_error) {
+      case X509_V_ERR_APPLICATION_VERIFICATION:
       case X509_V_ERR_CERT_CHAIN_TOO_LONG:
       case X509_V_ERR_CERT_HAS_EXPIRED:
       case X509_V_ERR_CERT_REVOKED:
       case X509_V_ERR_DEPTH_ZERO_SELF_SIGNED_CERT:
+      case X509_V_ERR_INVALID_PURPOSE:
       case X509_V_ERR_SELF_SIGNED_CERT_IN_CHAIN:
       case X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY:
       case X509_V_ERR_UNABLE_TO_VERIFY_LEAF_SIGNATURE:
-      case X509_V_ERR_APPLICATION_VERIFICATION:
         tls_log("client certificate failed verification: %s",
           X509_verify_cert_error_string(ctx_error));
         ok = 0;
         break;
 
-      case X509_V_ERR_INVALID_PURPOSE: {
-        register int i;
-        int count;
-
-        tls_log("client certificate failed verification: %s",
-          X509_verify_cert_error_string(ctx_error));
-
-        count = X509_PURPOSE_get_count();
```

Comment:
This particular scenario, of `X509_V_ERR_INVALID_PURPOSE`, can happen for multiple reasons.  (The one I encountered was from using `nsComment` in my OpenSSL config file, from a legacy configuration; I suspect that using other such legacy Netscape, or other, extensions could result in the same error, for no reason.)

In this particular case, logging the _possible_ purpose count (and what they are) is not useful, for such logging does _not_ indicate which particular purpose (or, most likely, _extension_) in the certificate caused OpenSSL such indigestion.  So I remove this logging entirely.